go/vt/vtgate/planbuilder/ddl.go

```diff
@@ -1,60 +1,91 @@
 package planbuilder
 
 import (
-	"fmt"
+	"vitess.io/vitess/go/vt/proto/vtrpc"
+	"vitess.io/vitess/go/vt/vterrors"
+	"vitess.io/vitess/go/vt/vtgate/vindexes"
 
 	"vitess.io/vitess/go/vt/key"
 	"vitess.io/vitess/go/vt/schema"
 	"vitess.io/vitess/go/vt/sqlparser"
 	"vitess.io/vitess/go/vt/vtgate/engine"
 )
 
-func buildDDLPlan(sql string, in sqlparser.Statement, vschema ContextVSchema) (engine.Primitive, error) {
-	stmt := in.(*sqlparser.DDL)
-	// This method call will validate the destination != nil check.
-	destination, keyspace, _, err := vschema.TargetDestination(stmt.Table.Qualifier.String())
-	if err != nil {
-		return nil, err
+func buildDDLPlan(sql string, stmt sqlparser.DDLStatement, vschema ContextVSchema) (engine.Primitive, error) {
+	var table *vindexes.Table
+	var destination key.Destination
+	var keyspace *vindexes.Keyspace
+	var err error
+
+	switch stmt.(type) {
+	case *sqlparser.CreateIndex:
+		// For Create index, the table must already exist
+		// We should find the target of the query from this tables location
+		table, _, _, _, destination, err = vschema.FindTableOrVindex(stmt.GetTable())
+		keyspace = table.Keyspace
+		if err != nil {
+			return nil, err
+		}
+		if table == nil {
+			return nil, vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "table does not exists: %s", stmt.GetTable().Name.String())
+		}
+		stmt.SetTable("", table.Name.String())
```

Comment:
for unsharded keyspace, most likely the Vschema would not contain the table information. So it might be possible that FindTableOrVindex would not return any table. in case table == nil, should check if the default keyspace or the qualifier is unsharded and pass that down to mysql.

Reply:
I have added a unit test for it, but it is already working correctly. The unsharded keyspace case is handled in https://github.com/vitessio/vitess/blob/4f7714f853b13913f4f32ff0ca17cd0e00a22ad1/go/vt/vtgate/vindexes/vschema.go#L494